xml/GenCode.php

```diff
@@ -14,7 +14,8 @@
 }
 
 if ($memLimit >= 0 and $memLimit < 536870912) {
-  ini_set('memory_limit', '512M');
```

Comment:
This approach is a bit strange - not sure why they bothered putting in a check for the existing `memory_limit`. Guess: in some environment, you're not allowed to call `ini_set('memory_limit')`, and this approach has the effect of avoiding an unnecessary call to `ini_set()`. But that's a total stab-in-the-dark.

Really tempted to just delete roughly lines 2-19 and simply call `ini_set('memory_limit', -1);` unconditionally.